### ISSUE TITLE
'/warnaserror-:DiagnosticId' should not nullify diagnostic severity settings for 'DiagnosticId' in editorconfig

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#49446]
**Version Used**: 

**Steps to Reproduce**:

1. Create a C# class library project with `net5` target framework with below code:
```csharp
class C
{
    void M() { }
}
```
2. Verify IDE suggestion/message [CA1822: Mark members as static](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822) is reported for method 'M'
3. Add an editorconfig file to the project root with following contents:
```editorconfig
[*.cs]
dotnet_diagnostic.CA1822.severity = warning
```
4. Build and verify `CA1822` warning
5. Edit the csproj file to enable `/warnaserror`:
```xml
<PropertyGroup>
  <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
</PropertyGroup>
```
6. Rebuild project and verify `CA1822` is reported as an error
7. Edit the csproj file to add `/warnaserror-:CA1822`:
```xml
<PropertyGroup>
  <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  <WarningsNotAsErrors>CA1822</WarningsNotAsErrors>
</PropertyGroup>
```
8. Rebuild project

**Expected Behavior**:
`CA1822` is reported as a warning

**Actual Behavior**:
`CA1822` is not reported as a warning. `/warnaserror-:CA1822` seems to cause compiler fallback `CA1822` to its default severity `Info`/`Suggestion` and ignores editorconfig entry that promotes it to a warning.

**Comment by mavasani** [comment]
I have a fix for the issue, will send a PR soon.
